src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -264,6 +303,80 @@ impl<'a, C: Client<T>, T: Timestamp + Lattice> StorageControllerMut<'a, C, T> {
             .ok_or(StorageError::IdentifierMissing(id))
     }
 
+    /// Persist timestamp bindings updates received from ingestion workers
+    pub(super) async fn persist_timestamp_bindings(
+        &mut self,
+        feedback: &TimestampBindingFeedback<T>,
+    ) {
+        for (id, bindings) in &feedback.bindings {
+            let mut ts_binding_stash = self
+                .storage
+                .stash
+                .collection::<PartitionId, ()>(&id.to_string())
+                .unwrap();
+
+            let collection_state = self.collection_mut(*id).expect("misisng_source_id");
+
+            let upper = ts_binding_stash.upper().unwrap();
+            let since = ts_binding_stash.since().unwrap();
+            let mut seal_frontier = upper.clone();
+
+            // Here we differentialize the bindings we got from workers
+            ts_binding_stash
+                .update_many(bindings.iter().cloned().flat_map(|(pid, ts, offset)| {
```

Comment:
I would either defensively sort `bindings` here, or put some ALLCAPS comments to the effect that it should be sorted by `ts`. It looks like that is necessary for correctess, but ignore if I'm wrong.

Reply:
no, stash doesn't require the operator to be sorted